source/geometries/LSCHallA.cc

```diff
@@ -91,16 +89,15 @@ namespace nexus {
     rock_region->SetProductionCuts(rock_prod_cut);
 
     // Basic sampling for inside of the walls (neutrons?)
-    G4ThreeVector hall_centre(0., -castle_centre_y_, -castle_centre_z_);
     hallA_vertex_gen_ =
       new CylinderPointSampler2020(lab_radius_ - 0.1 * mm,
 				   lab_radius_, (lab_length_ - 0.1 *mm) /2.,
-				   0., twopi, nullptr, hall_centre);
+				   0., pi, nullptr, G4ThreeVector(0., 0., 0.));
     hallA_outer_gen_ =
       new CylinderPointSampler2020(lab_radius_ + lab_wall_thickn_ + 0.1 * mm,
 				   lab_radius_ + lab_wall_thickn_ + 0.2 * mm,
 				   lab_length_/2. + lab_wall_thickn_ + 0.1 * mm,
-				   0, twopi, nullptr, hall_centre);
```

Comment:
The HallA geometry is not placed in `(0, 0, 0)`, therefore, if you centre the vertex generator in `(0, 0, 0)`, the vertices will not be generated correctly in the inside or outside of the rock, right?

Reply:
that's right, good catch